src/Facade/ResourceBuilders/PartsStorageTypeResourceBuilder.cs

```diff
@@ -15,7 +18,17 @@ public PartsStorageTypeResource Build(PartsStorageType partsStorageType, IEnumer
             return new PartsStorageTypeResource
                        {
                            StorageTypeCode = partsStorageType.StorageTypeCode,
+                           StorageType = new StorageTypeResource
+                           {
+                               StorageTypeCode = partsStorageType.StorageType.StorageTypeCode,
+                               Description = partsStorageType.StorageType.Description
+                           },
                            PartNumber = partsStorageType.PartNumber,
+                           Part = new PartResource 
+                            {
+                               PartNumber = partsStorageType.Part.PartNumber,
+                               Description = partsStorageType.Part.Description
+                            },
```

Comment:
both PartResources and StorageTypeResources have their own dedicated ResourceBuilders which you could make use of here